docs/designers-developers/developers/block-api/block-context.md

```diff
@@ -0,0 +1,70 @@
+# Block Context
+
+Block context is a feature which enables ancestor blocks to provide values which can be consumed by descendent blocks within its own hierarchy. Those descendent blocks can inherit this value without resorting to hard-coded values, or without an explicit awareness of the block which provides those values.
+
+This is especially useful in full-site editing where, for example, the contents of a block may depend on the context of the post in which it is relevant. A blogroll template may show excerpts of many different posts. Using block context, there can still be one single "Post Excerpt" block which displays the contents of the post based on an inherited post ID.
+
+If you are familiar with [React Context](https://reactjs.org/docs/context.html), the idea is very similar. In fact, the client-side block editor implementation of block context is a very simple application of React Context. Block context is also supported in server-side `render_callback` implementations, demonstrated in the examples below.
+
+## Defining Block Context
+
+Block context is defined in the registered settings of a block. A block can provide a context value, or consume a value it seeks to inherit.
+
+### Providing Block Context
+
+A block can provide a context value by assigning a `providesContext` property in its registered settings. This should be assigned as an array of the attribute names for attribute values to assign as context. Currently, block context only supports values derived from the block's own attributes. This could be enhanced in the future to support additional sources of context values.
+
+_block.json_
+
+```json
+{
+	"name": "core/post",
+	"attributes": {
+		"postId": {
+			"type": "number"
+		}
+	},
+	"providesContext": [ "postId" ]
+}
+```
+
+### Consuming Block Context
+
+A block can inherit a context value from an ancestor provider by assigning a `context` property in its registered settings. This should be assigned as an array of the context names the block seeks to inherit.
+
+_block.json_
+
+```json
+{
+	"name": "core/post-excerpt",
+	"context": [ "postId" ]
```

Comment:
`usesContext` could be a better name to go with `providesContext`. When I saw `context` on its own later in the pull request I had to remind myself that it was consuming instead of providing.

Reply:
> `usesContext` could be a better name to go with `providesContext`. When I saw `context` on its own later in the pull request I had to remind myself that it was consuming instead of providing.

I'm leaning toward keeping the current name. I still don't feel particularly strongly about this, but prefer the name for reasons given above. There is also some prior art in the inherited React similarities. In the [Legacy API](https://reactjs.org/docs/legacy-context.html), this relationship was between `childContextTypes` and `contextTypes`, similarly leaning into the ancestor as providing the qualifier ("child", "provides"), leaving free the base name for the majority-case consumer.